src/components/dashboard/Claim.js

```diff
@@ -68,15 +68,24 @@ const Claim = props => {
     const isValid = screenProps.screenState && screenProps.screenState.isValid
 
     log.debug('from FR:', { isValid })
-
-    if (isValid && (await goodWallet.isCitizen())) {
-      handleClaim()
-    } else if (isValid === false) {
-      screenProps.goToRoot()
-    } else {
-      if (isCitizen === false) {
-        goodWallet.isCitizen().then(_ => gdstore.set('isLoggedInCitizen')(_))
+    try {
+      if (isValid && (await goodWallet.isCitizen())) {
+        handleClaim()
+      } else if (isValid === false) {
+        screenProps.goToRoot()
+      } else {
+        if (isCitizen === false) {
+          goodWallet.isCitizen().then(_ => gdstore.set('isLoggedInCitizen')(_))
+        }
       }
+    } catch (e) {
+      log.error('evaluateFRValidity failed', e.message, e)
+      showErrorDialog('Something unexpected happened', '', {
+        boldMessage: 'Try again.',
```

Comment:
@AlexeyKosinski does this conform to the usual format of error messages?

Reply:
@StanislavShevchenko please change this error message to: "The error happens in blockchain request, please try again later"